README.md

```diff
@@ -18,25 +20,184 @@ $ npm install --save pipage
+
+```js
+// A blank pass-through pipeline:
+var pipeline = new Pipage()
+
+// Initialized with already existing streams:
+var pipeline = new Pipage([ transform1, transform2, ... ])
+```
+
+### Adding streams
+
+```js
+// Streams can be appended,
+pipeline.append( stream )
+pipeline.append( stream1, stream2, ..., streamN )
+
+// prepended,
+pipeline.prepend( stream1, stream2, ..., streamN )
+
+// inserted at a specific index,
+pipeline.insert( 3, stream1, ..., streamN )
+
+// which is the same as splicing in streams (add N streams at index 3):
+pipeline.splice( 3, 0, stream1, ..., streamN )
+```
+
+### Removing streams
+
+```js
+// Streams can be shifted off the beginning,
+var firstStream = pipeline.shift()
+
+// or popped off the end,
+var lastStream = pipeline.pop()
+
+// spliced out at a specific index (remove 3 from index 2),
+var removedStreams = pipeline.splice( 2, 3 )
+
+// or removed by reference:
+pipeline.remove( stream )
+```
+
+### Selecting streams
+
+```js
+// Get a stream at a specific index in the pipeline:
+var stream = pipeline.get( 2 )
+var lastStream = pipeline.get( -1 )
+
+// Find a stream in the pipeline:
+var index = pipeline.indexOf( stream )
+var lastIndex = pipeline.lastIndexOf( stream )
+```
+
+### Example
+
+```js
+var Pipage = require('pipage')
+var path = require('path')
+var zlib = require('zlib')
+var unbzip2 = require('unbzip2-stream')
+var xz = require('xz')
+
+var pipeline = new Pipage()
+
+switch( path.extname( filename ) ) {
+  case '.gz':
+    pipeline.prepend( zlib.createUnzip() );
+    break
+  case '.bz':
+  case '.bz2':
+    pipeline.prepend( unbzip2() );
+    break
+  case '.xz':
+    pipeline.prepend( new xz.Decompressor() )
+    break
+}
+
+fs.createReadStream( filename )
+  .pipe( pipeline )
+  .pipe( fs.createWriteStream( destination ) )
+```
+
+### Events
+
+#### Error handling
+
+Errors from streams within the pipeline are listened to, and re-emitted on the pipeline itself, with an additional `.stream` property being set on the `error` object, pointing to the stream which emitted it:
+
+```js
+pipeline.on( 'error', function( error ) {
+  // This error originated from the pipeline's internal stream
+  // pointed to by `error.stream`, not from the pipeline itself
+  if( error.stream ) {
+    // ...
+  } else {
+    // This error came from the pipeline itself
+  }
+})
+```
+
+#### Binding to contained stream's events
+
 ```js
-var Pipeline = require( 'pipage' )
+var stream = pipeline.get(-1)
+
+// Re-emit a stream's events on the pipeline:
+pipeline.bind( stream, 'eventname' )
+pipeline.bind( stream, [ 'someevent', 'otherevent' ])
+
+// Remove re-emission of a stream's event:
+pipeline.unbind( stream, 'eventname' )
+
+// Stop re-emission of all of stream's events on the pipeline:
+pipeline.unbindAll( stream )
 ```
 
+##### Example
+
 ```js
-var checksum = new ChecksumStream( metadata )
-var slice = new SliceStream( metadata )
+var Pipage = require('pipage')
+var crypto = require('crypto')
```

Comment:
Yeah, that's a leftover from when I had a sample checksum stream in here. Removed.